src/sentry/api/endpoints/release_threshold_status_index.py

```diff
@@ -0,0 +1,182 @@
+                        project,
+                        ...,
+                        is_healthy: True/False
+                    },
+                    {...},
+                ],
+                project2_id: [],
+                ...
+            },
+            ...
+        }
+        ``````````````````
+
+        :param start: timestamp of the beginning of the specified date range
+        :param end: timestamp of the end of the specified date range
+
+        TODO:
+        - should we limit/paginate results? (this could get really bulky)
+        """
+        data = request.data if len(request.GET) == 0 and hasattr(request, "data") else request.GET
+        start, end = get_date_range_from_params(params=data)
+
+        serializer = ReleaseThresholdStatusIndexSerializer(
+            data=request.query_params,
+        )
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        environments_list = serializer.validated_data.get("environment")
+        project_ids_list = serializer.validated_data.get("project")
+        release_ids_list = serializer.validated_data.get("release_id")
+
+        # NOTE: we're only filtering on date ADDED
+        # This is not synonymous with a deploy... which may be what we actually want.
+        release_query = Q(organization=organization, date_added__gte=start, date_added__lte=end)
+        environments = self.get_environments(request, organization)
+        if environments:
+            environments_list = [env.name for env in environments]
+            release_query &= Q(
+                releaseprojectenvironment__environment__name__in=environments_list,
+            )
+        if project_ids_list:
+            release_query &= Q(
+                projects__id__in=project_ids_list,
+            )
+        if release_ids_list:
+            release_query &= Q(
+                id__in=release_ids_list,
+            )
+
+        queryset = (
+            Release.objects.filter(release_query)
+            .annotate(
+                date=F("date_added"),  # transforms date_added into 'date'
+            )
+            .order_by("-date")
+            .distinct()
+        )
+        queryset.prefetch_related("projects__release_thresholds")
+
+        # ========================================================================
+        # TODO:
+        # Fetch relevant snuba/sessions data
+        # Determine which thresholds have succeeded/failed
+
+        release_threshold_health: DefaultDict[int, Any] = defaultdict()
+        for release in queryset:
+            release_project = defaultdict(list)
+            if project_ids_list:
+                project_list = release.projects.filter(id__in=project_ids_list)
+            else:
+                project_list = release.projects.all()
+            for project in project_list:
+                project_threshold_statuses = []
+                for threshold in project.release_thresholds.all():
```

Comment:
i think you should filter for the `environment` on the threshold too if you query for the `environment`

Reply:
Good call
updated